core/services/ocr2/plugins/ocr2keeper/evm21/transmit/event_provider.go

```diff
@@ -147,70 +154,26 @@ func (c *TransmitEventProvider) GetLatestEvents(ctx context.Context) ([]ocr2keep
 		return nil, fmt.Errorf("%w: failed to collect logs from log poller", err)
 	}
 
-	parsed, err := c.parseLogs(logs)
-	if err != nil {
-		return nil, fmt.Errorf("%w: failed to parse logs", err)
-	}
-
-	return c.convertToTransmitEvents(parsed, end)
+	return c.processLogs(end, logs...)
 }
 
-func (c *TransmitEventProvider) parseLogs(logs []logpoller.Log) ([]transmitEventLog, error) {
-	results := []transmitEventLog{}
+// processLogs will parse the unseen logs and return the corresponding transmit events.
+// If a log was seen before it will be picked from the cache.
```

Comment:
I think we should also be ok in skipping a already processed log and not returning it at all in `GetLatestEvents` but can explore that optimisation in another PR if still needed